src/controllers/report-controller.js

```diff
@@ -0,0 +1,133 @@
+        .populate({path: 'accountReported', select: 'name lastname -_id'})
+        .then(function (reports) {  
+            if(reports.length){
+                response.status(StatusCodes.OK).json(reports);
+            }
+            else{
+                responseNotFound(response);
+            }
+        })
+        .catch(function (error){
+            responseServer(response, error);
+        }); 
+    }
+    else{
+        let queryAccountReported = null;
+        switch(filter){
+            case "nameAccount":
+                queryAccountReported = {$match:{'idAccount.name': { $regex : new RegExp(criterion, "i")}}};
+                break;
+            case "lastnameAccount":
+                queryAccountReported = {$match:{'idAccount.lastname': { $regex : new RegExp(criterion, "i")}}};
+                break;
+            case "nameReported":
+                queryAccountReported = {$match:{'accountReported.name': { $regex : new RegExp(criterion, "i")}}};
+                break;
+            case "lastnameReported":
+                queryAccountReported = {$match:{'accountReported.lastname': { $regex : new RegExp(criterion, "i")}}};
+                break;
+        }
+        Reports.aggregate([
+            {$lookup: {
+                from:'accounts', as:'idAccount',
+                localField:'idAccount',
+                foreignField:'_id',
+            },
+            }, 
+            {$lookup: {
+                from:'accounts', as:'accountReported',
+                localField:'accountReported',
+                foreignField:'_id',
+            },
+            }, 
+            queryAccountReported,
+            {$project:{'accountReported.name': 1, 'accountReported.lastname': 1, dateCreation:1, 'idAccount.name': 1, 
+            'idAccount.lastname': 1}}
+        ])
+        .then(function (reports) {  
+            if(reports.length){
+                response.status(StatusCodes.OK).json(reports);
+            }
+            else{
+                responseNotFound(response);
+            }
+        })
+        .catch(function (error){
+            responseServer(response, error);
+        });
+    }
+}
+
+
+const getReports = async (request, response) => {
+    Reports.find(null, {_id:0, dateCreation:1})
+    .populate({path: 'idAccount', select: 'name lastname -_id'})
+    .populate({path: 'accountReported', select: 'name lastname -_id'})
+    .then(function (reports) {  
+        if(reports.length){
+            response.status(StatusCodes.OK).json(reports);
+        }
+        else{
+            responseNotFound(response);
+        }
+    })
+    .catch(function (error){
+        responseServer(response, error);
+    });
+}
+
+
+const getReport = async (request, response) => {
+    const reportID = request.params.reportID;
+    Reports.findById(reportID)
+    .populate({path: 'idAccount', select: 'name lastname username -_id'})
+    .populate({path: 'accountReported', select: 'name lastname username -_id'})
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Martha-Miroslava/APIBiziidiDiidxaza/pullRequest?prid=8267260)
```suggestion
    .populate({path: "accountReported", select: 'name lastname username -_id'})
```